pkgs/development/libraries/science/math/mkl/default.nix

```diff
@@ -82,4 +26,77 @@ stdenvNoCC.mkDerivation rec {
     platforms = [ "x86_64-linux" "x86_64-darwin" ];
     maintainers = [ maintainers.bhipple ];
   };
-}
+
+  mkl = stdenvNoCC.mkDerivation rec {
+    name = "mkl-${version}";
+
+    src = if stdenvNoCC.isDarwin
+      then
+        (fetchurl {
+          url = "http://registrationcenter-download.intel.com/akdlm/irc_nas/tec/13565/m_mkl_${version}.dmg";
+          sha256 = "1f1jppac7vqwn00hkws0p4njx38ajh0n25bsjyb5d7jcacwfvm02";
+        })
+      else
+        (fetchurl {
+          url = "http://registrationcenter-download.intel.com/akdlm/irc_nas/tec/13575/l_mkl_${version}.tgz";
+          sha256 = "1bf7i54iqlf7x7fn8kqwmi06g30sxr6nq3ac0r871i6g0p3y47sf";
+        });
+
+    buildInputs = if stdenvNoCC.isDarwin then [ undmg ] else [ rpmextract ];
+
+    buildPhase = if stdenvNoCC.isDarwin then ''
+        for f in Contents/Resources/pkg/*.tgz; do
+            tar xzvf $f
+        done
+    '' else ''
+      rpmextract rpm/intel-mkl-common-c-${date}-${rel}-${date}-${rel}.noarch.rpm
+      rpmextract rpm/intel-mkl-core-rt-${date}-${rel}-${date}-${rel}.x86_64.rpm
+    '';
+
+    installPhase = if stdenvNoCC.isDarwin then ''
+        mkdir -p $out/lib
+        cp -r compilers_and_libraries_${version}/mac/mkl/include $out/
+        cp -r compilers_and_libraries_${version}/mac/mkl/lib/* $out/lib/
+        cp -r compilers_and_libraries_${version}/licensing/mkl/en/license.txt $out/lib/
+    '' else ''
+        mkdir -p $out/lib
+        cp -r opt/intel/compilers_and_libraries_${version}/linux/mkl/include $out/
+        cp -r opt/intel/compilers_and_libraries_${version}/linux/mkl/lib/intel64_lin/* $out/lib/
+        cp license.txt $out/lib/
+    '';
+
+    # Per license agreement, do not modify the binary
+    dontStrip = true;
+    dontPatchELF = true;
+
+    # Since these are unmodified binaries from Intel, they do not depend on stdenv
+    # and we can make them fixed-output derivations for cache efficiency.
+    outputHashAlgo = "sha256";
+    outputHashMode = "recursive";
+    outputHash = if stdenvNoCC.isDarwin
+      then "1224dln7n8px1rk8biiggf77wjhxh8mzw0hd8zlyjm8i6j8w7i12"
+      else "0d8ai0wi8drp071acqkm1wv6vyg12010y843y56zzi1pql81xqvx";
+
+    inherit meta;
+  };
+
+  libvar = if stdenvNoCC.isDarwin then "DYLD_LIBRARY_PATH" else "LD_LIBRARY_PATH";
+in
```

Comment:
I suppose this is unfriendly to overlays being able to override `mkl` versions easily.  Is there an idiomatic way to proceed?  Pass all the versions and fixed-output shas as defaulted parameters?  Seems clunky enough that I bet a user who wants to fork it will just copy the expression.

Reply:
Why not just expose unwrapped `mkl` as attribute in `pkgs`? Then it can be overridden like any other package.

Reply:
Makes sense; updated accordingly!